v2/runners/coild_server.go

```diff
@@ -189,83 +192,172 @@ func (s *coildServer) Add(ctx context.Context, args *cnirpc.CNIArgs) (*cnirpc.Ad
 	}
 	return &cnirpc.AddResponse{Result: data}, nil
 }
 
+func setCoilInterfaceAlias(interfaces map[string]bool, conf *nodenet.PodNetConf, logger *zap.Logger, pod *corev1.Pod) error {
+	ifName := ""
+	for name, isSandbox := range interfaces {
+		if !isSandbox {
+			ifName = name
+			break
+		}
+	}
+	logger.Sugar().Infof("interface selected: %s", ifName)
+	hLink, err := netlink.LinkByName(ifName)
+	if err != nil {
+		return fmt.Errorf("netlink: failed to look up the host-side veth: %w", err)
+	}
+	logger.Sugar().Infof("link found: %v", hLink)
+
+	// give identifer as an alias of host veth
+	if err := netlink.LinkSetAlias(hLink, nodenet.GenAlias(conf, string(pod.UID))); err != nil {
+		return fmt.Errorf("netlink: failed to set alias: %w", err)
+	}
+	return nil
+}
+
+func getPodIPs(ips string) (net.IP, net.IP) {
+	var ipv4, ipv6 net.IP
+	podIPs := strings.Split(ips, ",")
+	for _, ip := range podIPs {
+		addr := net.ParseIP(ip)
+		if addr != nil {
+			if ipv4 == nil && addr.To4() != nil {
+				ipv4 = addr
+			} else if ipv6 == nil {
+				ipv6 = addr
+			}
+		}
+		if ipv4 != nil && ipv6 != nil {
+			break
+		}
+	}
+	return ipv4, ipv6
+}
+
 func (s *coildServer) Del(ctx context.Context, args *cnirpc.CNIArgs) (*emptypb.Empty, error) {
 	logger := withCtxFields(ctx, s.logger)
 
-	if err := s.podNet.Destroy(args.ContainerId, args.Ifname); err != nil {
-		logger.Sugar().Errorw("failed to destroy pod network", "error", err)
-		return nil, newInternalError(err, "failed to destroy pod network")
+	ipamEnabled, _, err := getSettings(args)
+	if err != nil {
+		return nil, newInternalError(err, "error parsing settings")
 	}
 
-	if err := s.nodeIPAM.Free(ctx, args.ContainerId, args.Ifname); err != nil {
-		logger.Sugar().Errorw("failed to free addresses", "error", err)
-		return nil, newInternalError(err, "failed to free addresses")
+	if ipamEnabled {
+		if err := s.podNet.Destroy(args.ContainerId, args.Ifname); err != nil {
+			logger.Sugar().Errorw("failed to destroy pod network", "error", err)
+			return nil, newInternalError(err, "failed to destroy pod network")
+		}
+
+		if err := s.nodeIPAM.Free(ctx, args.ContainerId, args.Ifname); err != nil {
+			logger.Sugar().Errorw("failed to free addresses", "error", err)
+			return nil, newInternalError(err, "failed to free addresses")
+		}
 	}
 	return &emptypb.Empty{}, nil
 }
 
 func (s *coildServer) Check(ctx context.Context, args *cnirpc.CNIArgs) (*emptypb.Empty, error) {
 	logger := withCtxFields(ctx, s.logger)
 
-	if err := s.podNet.Check(args.ContainerId, args.Ifname); err != nil {
-		logger.Sugar().Errorw("check failed", "error", err)
+	ipamEnabled, _, err := getSettings(args)
+	if err != nil {
 		return nil, newInternalError(err, "check failed")
 	}
+
+	if ipamEnabled {
+		if err := s.podNet.Check(args.ContainerId, args.Ifname); err != nil {
+			logger.Sugar().Errorw("check failed", "error", err)
+			return nil, newInternalError(err, "check failed")
+		}
+	}
+
```

Comment:
I've added basic check for interface alias (as it was there before my changes and it was miising now in egress-only mode).
I was thinking about adding further checks, but it seems that to use e.g. IsInitalized functions of `founat` package we would need to get pod, service and egress from KubeAPI each time check is called, so it would need to be considered if this is worth it.
I think, that additional checks could be added, but I would rather have those as separate PR as this one is already kinda big.

Reply:
Thanks.
I agree that we consider this later.
I think coil's CNI check command needs more consideration(not only egress-only mode, also ipam mode).
Therefore, this implementation is enough for now.